src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -54,65 +55,125 @@ public async Task UpdateSecretsManagerSubscription(SecretsManagerSubscriptionUpd
         }
 
-        if (update.ServiceAccountsAdjustment != 0)
+        if (!organization.UseSecretsManager)
         {
-            await AdjustServiceAccountsAsync(organization, update, plan);
+            throw new BadRequestException("Organization has no access to Secrets Manager.");
         }
+    }
 
-        if (update.MaxAutoscaleSeats != organization.MaxAutoscaleSmSeats)
+    private async Task FinalizeSubscriptionAdjustmentAsync(Organization organization,
+        Plan plan, Tuple<int, int>? adjustSeatsResponse, Tuple<int, int>? adjustServiceAccountsResponse)
+    {
+        if (adjustSeatsResponse != null)
         {
-            UpdateSeatsAutoscaling(organization, update.MaxAutoscaleSeats, plan);
+            if (adjustSeatsResponse.Item1 != 0)
+            {
+                await ProcessChargesAndRaiseEventsForAdjustSeatsAsync(organization, plan, adjustSeatsResponse.Item1,
+                    adjustSeatsResponse.Item2);
+            }
         }
 
-        if (update.MaxAutoscaleServiceAccounts != organization.MaxAutoscaleSmServiceAccounts)
+        if (adjustServiceAccountsResponse != null)
         {
-            UpdateServiceAccountAutoscaling(organization, update.MaxAutoscaleServiceAccounts, plan);
+            if (adjustServiceAccountsResponse.Item1 != 0)
+            {
+                await ProcessChargesAndRaiseEventsForAdjustServiceAccountsAsync(organization, plan,
+                    adjustServiceAccountsResponse.Item1,
+                    adjustServiceAccountsResponse.Item2);
+            }
         }
-
         await _organizationService.ReplaceAndUpdateCacheAsync(organization);
+    }
 
+    private async Task SendEmailIfAutoscaleLimitReached(Organization organization)
+    {
         if (organization.SmSeats.HasValue && organization.MaxAutoscaleSmSeats.HasValue && organization.SmSeats == organization.MaxAutoscaleSmSeats)
         {
-            await SendEmailAsync(organization, organization.MaxAutoscaleSmSeats.Value, "Seats");
+            await SendSeatLimitEmailAsync(organization, organization.MaxAutoscaleSmSeats.Value, "Seats");
         }
 
         if (organization.SmServiceAccounts.HasValue && organization.MaxAutoscaleSmServiceAccounts.HasValue && organization.SmServiceAccounts == organization.MaxAutoscaleSmServiceAccounts)
         {
-            await SendEmailAsync(organization, organization.MaxAutoscaleSmServiceAccounts.Value, "Service Accounts");
+            await SendServiceAccountLimitEmailAsync(organization, organization.MaxAutoscaleSmServiceAccounts.Value, "Service Accounts");
+        }
+    }
+
+    private async Task SendSeatLimitEmailAsync(Organization organization, int MaxAutoscaleValue, string adjustingProduct)
+    {
+        try
+        {
+            var ownerEmails = (await _organizationUserRepository.GetManyByMinimumRoleAsync(organization.Id,
+                    OrganizationUserType.Owner))
+                .Select(u => u.Email).Distinct();
```

Comment:
We could simply add a check if there are any emails before sending `if (ownerEmails.Any())`

Reply:
I think it's probably fine, because I don't think providers are intended to manage SM. Worst case it gets handled by the try/catch. I'll make a note to follow up if required.